### ISSUE TITLE
consider  making `totag` a Nunjucks template

### ISSUE THREAD

**Issue by Iced-Sun** [lizheming/drone-wechat#4]
Occasionally, notifications would be better to be sent to different groups according to the branch/build status/event type, for the sake of preventing bothering irrelevant people.

Currently, multiple steps with different conditions have to be set to achieve that. I think it may be a good idea to take `to_tag` as another Nunjucks template, with a simple modification of source code: `totag` -> `render(totag)`.

Regards.

**Comment by lizheming** [comment]
Did you want send notification difference between pull request build end and normal commit build end? If we add template render feature to `totag` parameter, I think we should also add the feature to `toparty` and `touser`.

**Comment by Iced-Sun** [comment]
Possible scenarios could be:
1. if the building fails, only the involved development staff would be notified, rather than bothering the testing/managing groups;
2. if a staging/testing branch is successfully built, the testing/QA team would be notified;
3. if a tag/release is successfully built, the managing/product team would be notified;
4. if a development branch is successfully built, the development staff would be notified.

IMHO, `toparty` and `touser` could be left untouched since a tag could be dynamically assigned to each user, while a party/user is relatively fixed. Under the above scenario, the fixed part/user destination is less flexible.

Regards.

**Comment by lizheming** [comment]
OK, I understand it. But I think you can use `when` keyword with multiple step in pipeline to achieve it. https://docs.drone.io/user-guide/pipeline/conditions/

```yaml
steps:
- name: Notify to developer
  image: lizheming/drone-wechat
  ...
  when:
    status:
      - success
      - failure
- name: Notify to QA
  image: lizheming/drone-wechat
  ...
  when:
    branch:
       - testing
    status:
      - success
- name: Notify to PM
  image: lizheming/drone-wechat
  ...
  when:
    event:
      - tag
    status:
      - success
```

**Comment by Iced-Sun** [comment]
Surely, that's what I do now.

The whole point of a dynamic `to_tag` is to avoid multiple 'when'.

Regards.

**Comment by lizheming** [comment]
OK, I'll add this feature later.

**Comment by lizheming** [comment]
drone-wechat@1.1.6 released, you can use `1.1.6` or `latest` tag to use the feature.

**Comment by Iced-Sun** [comment]
Lots of thanks!
